router/throttler/throttler.go

```diff
@@ -0,0 +1,156 @@
+package throttler
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/coinpaprika/ratelimiter"
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+)
+
+var (
+	destSettingsMap map[string]Settings
+)
+
+//Throttler is an interface for throttling functions
+type Throttler interface {
+	LimitReached(key string) bool
+	IsEnabled() bool
+}
+
+type Limiter struct {
+	enabled     bool
+	eventLimit  int
+	timeWindow  time.Duration
+	ratelimiter *ratelimiter.RateLimiter
+}
+
+type Settings struct {
+	eventLimit             int
+	timeWindowInS          int
+	userLevelThrottling    bool
+	userLevelLimit         int
+	userLevelTimeWindowInS int
+}
+
+//HandleT is a Handle for event limiter
+type HandleT struct {
+	destinationName string
+	destLimiter     *Limiter
+	userLimiter     *Limiter
+}
+
+func init() {
+	loadConfig()
+}
+
+func loadConfig() {
+	destSettingsMap = map[string]Settings{
+		// https://customer.io/docs/api/#api-documentationlimits
+		"CUSTOMERIO": {
+			eventLimit:          30,
+			timeWindowInS:       1,
+			userLevelThrottling: false,
+		},
+		// https://help.amplitude.com/hc/en-us/articles/360032842391-HTTP-API-V2#upload-limit
+		"AM": {
+			eventLimit:             1000,
+			timeWindowInS:          1,
+			userLevelThrottling:    true,
+			userLevelLimit:         10,
+			userLevelTimeWindowInS: 1,
+		},
+	}
```

Comment:
Move this map to some other file. Looks clean that way.

No need to do it now. We can do it when we add throttling info for more destination.